### ISSUE TITLE
False positive: unsubscriptable-object when using classmethod and property together

### ISSUE THREAD

**Issue by BelchiorSaeg** [PyCQA/pylint#5699]
### Bug description

```
from typing import List


class Unknown:

    CONSTANT_1 = 0
    CONSTANT_2 = 1
    CONSTANT_3 = 2

    @classmethod
    @property
    def all_constants(cls) -> List[int]:
        return [cls.CONSTANT_1, cls.CONSTANT_2, cls.CONSTANT_3]


print(Unknown.all_constants[0])  # pylint: disable=unsubscriptable-object
```
`... 0`

```
>>> Unknown.all_constants
... [0, 1, 2]
```

### Command used

```shell
pylint untitled.py and VSCode linter
```

### Pylint output

```shell
************* Module untitled
untitled.py:16:6: E1136: Value 'Unknown.all_constants' is unsubscriptable (unsubscriptable-object)

-------------------------------------------------------------------
Your code has been rated at -1.25/10
```

### Expected behavior

************* Module untitled

------------------------------------------------------------------
Your code has been rated at 10/10 

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.2
Python 3.9.9 | packaged by conda-forge | (main, Dec 20 2021, 02:36:06) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

> **Edição	Windows** 11 Pro
> **Versão**	21H2
> **Instalado em**	‎10/‎10/‎2021
> **Compilação do SO**	22000.434
> **Experiência	Pacote de Experiência de Recursos do Windows** 1000.22000.434.0
>
